File: backend/src/models/Therapist.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Therapist = sequelize.define('Therapist', {
    id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true,
    },
    user_id: {
      type: DataTypes.UUID,
      allowNull: false,
      unique: true,
      references: {
        model: 'users',
        key: 'id',
      },
    },
    license_number: {
      type: DataTypes.STRING(50),
      allowNull: false,
      validate: {
        notEmpty: { msg: 'License number is required' },
      },
    },
    license_doc_url: {
      type: DataTypes.STRING(500),
      allowNull: true,
    },
    photo_url: {
      type: DataTypes.STRING(500),
      allowNull: true,
    },
    specialization: {
      type: DataTypes.STRING(100),
      allowNull: true,
    },
    status: {
      type: DataTypes.ENUM('pending', 'active', 'suspended'),
      defaultValue: 'pending',
      allowNull: false,
    },
    validated_by: {
      type: DataTypes.UUID,
      allowNull: true,
      references: {
        model: 'users',
        key: 'id',
      },
    },
    validated_at: {
      type: DataTypes.DATE,
      allowNull: true,
    },
    rating: {
      type: DataTypes.DECIMAL(3, 2),
      defaultValue: 0.00,
      validate: {
        min: 0,
        max: 5,
      },
    },
  }, {
    tableName: 'therapists',
    timestamps: true,
    underscored: true,
    indexes: [
      { unique: true, fields: ['user_id'] },
      { fields: ['status'] },
      { fields: ['specialization'] },
    ],
  });

  return Therapist;
};

File: mobile/lib/models/therapist.dart
import 'user.dart';

class Therapist {
  final String id;
  final String userId;
  final String licenseNumber;
  final String? licenseDocUrl;
  final String? photoUrl;
  final String? specialization;
  final String status;
  final String? validatedBy;
  final DateTime? validatedAt;
  final double rating;
  final User? user;

  Therapist({
    required this.id,
    required this.userId,
    required this.licenseNumber,
    this.licenseDocUrl,
    this.photoUrl,
    this.specialization,
    required this.status,
    this.validatedBy,
    this.validatedAt,
    this.rating = 0.0,
    this.user,
  });

  factory Therapist.fromJson(Map<String, dynamic> json) {
    return Therapist(
      id: json['id'] ?? '',
      userId: json['user_id'] ?? '',
      licenseNumber: json['license_number'] ?? '',
      licenseDocUrl: json['license_doc_url'],
      photoUrl: json['photo_url'],
      specialization: json['specialization'],
      status: json['status'] ?? 'pending',
      validatedBy: json['validated_by'],
      validatedAt:
          json['validated_at'] != null
              ? DateTime.parse(json['validated_at'])
              : null,
      rating: double.tryParse(json['rating']?.toString() ?? '0') ?? 0.0,
      user: json['user'] != null ? User.fromJson(json['user']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'license_number': licenseNumber,
      'license_doc_url': licenseDocUrl,
      'photo_url': photoUrl,
      'specialization': specialization,
      'status': status,
      'rating': rating,
    };
  }

  /// Display name (from nested user or fallback)
  String get displayName => user?.name ?? 'Terapis';
}
